sailthru/Sailthru_Client.php

```diff
@@ -81,6 +82,19 @@ public function  __construct($api_key, $secret, $api_uri = false) {
         }
 
         $this->http_request_type = function_exists('curl_init') ? 'httpRequestCurl' : 'httpRequestWithoutCurl';
+
+        if ($options) {
+            $this->options['timeout'] = $options['timeout'] ? $options['timeout'] : $this->options['timeout'];
+            $this->options['connect_timeout'] = $options['connect_timeout'] ? $options['connect_timeout'] : $this->options['connect_timeout'];
+        }
```

Comment:
If $options doesn't have a 'timeout' key or a 'connect_timeout' key, accessing them will throw a warning. Consider using isset instead.

Consider validating the inputs. Per http://php.net/manual/en/function.curl-setopt.php, curl expects an integer, so simply casting the value to an integer should suffice.

Should we add setters as well since we allow other things to be set (e.g. http headers)? Your call.

With only two options, maybe they should be individual parameters? I don't feel strongly about this one.